src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -168,16 +170,23 @@ def is_databricks_dataset(self) -> bool:
     def is_table_in_mount(self) -> bool:
         return self.database.startswith("mounted_") and self.is_delta
 
+    @property
+    def is_managed(self) -> bool:
+        return self.object_type == "MANAGED"
+
     @property
     def what(self) -> What:
+        dbfs_delta = {True: What.DBFS_ROOT_DELTA, False: What.DBFS_ROOT_NON_DELTA}
         if self.is_databricks_dataset:
             return What.DB_DATASET
+        if self.is_managed and self.is_table_in_mount:
+            return What.MANAGED_MOUNT
         if self.is_table_in_mount:
             return What.TABLE_IN_MOUNT
-        if self.is_dbfs_root and self.is_delta:
-            return What.DBFS_ROOT_DELTA
         if self.is_dbfs_root:
-            return What.DBFS_ROOT_NON_DELTA
+            return dbfs_delta[self.is_delta]
```

Comment:
rewrite as two separate `if` statements. it's quite confusing to read it under stress as it is now ;)